internal/app/tfsec/scanner/scanner.go

```diff
@@ -153,12 +154,43 @@ func (scanner *Scanner) checkRangeIgnored(id string, r block.Range, b block.Bloc
 	}
 
 	if foundValidIgnore {
-		return isIgnoreWithinExpiry(ignoreLine, id, b.Range())
+		return scanner.verifyIgnoreConditions(ignoreLine, id, b)
 	}
 
 	return foundValidIgnore
 }
 
+func (scanner *Scanner) verifyIgnoreConditions(ignoreLine string, id string, b block.Block) bool {
+	for _, ignoreCode := range strings.Split(ignoreLine, " ") {
+		if strings.Contains(ignoreCode, id) || strings.Contains(ignoreCode, "ignore:*") {
```

Comment:
I think this is going to match sometimes when it shouldn't like `ignore:aws-123` will match on `Contains(ignore:aws-1)`

Reply:
done